### DISCUSSION TITLE
Map my house?

### DISCUSSION THREAD

**Discussion by SurreyMichael** [ESPresense/ESPresense#1334] [Ideas]
I posted the following in the HA forum but got referred here.  Hope it might be of interest to someone.

I would love Espresense to work as a matrix rather than a set of individual nodes.

I use it to switch on/off my sonos speakers when I move around my house. Espresense nodes work really well, but sometimes I am nearer a node upstairs than a node across a room. Whilst the system functions well, the placement and adjustment of each node makes it rather limiting as they function independently.

It would be great if HA could combine distance measurements from nodes to map my house. In an ideal installation, I would have a few nodes in the house and a ‘learn mode’ during which HA maps the house according to relative distances to the nodes while I walk around to ‘teach’ it.

I tried creating sensors in HA to report distance measurements so that I could program this myself - but it turned out well beyond my capability to get further. The mapping would be, I guess, a pattern recognition challenge so maybe AI could help? The distances are not accurate but are consistent, so mapping should be possible.

Thanks for reading - maybe it triggers an idea for someone!

**Comment by SurreyMichael** [comment]
Oops - is this what ESPresense-companion does?  Apologies if I am asking for something that already exists!

**Reply by leccelecce** [reply]
Companion allows you to draw a multi-floor 2D layout of your house, place your ESPresence nodes on it, then triangulate your tracked devices with a confidence score.

It's pretty awesome, and there is a separate community tool to render this in 3D too.
